driver-reactive-streams/src/main/com/mongodb/reactivestreams/client/internal/FindPublisherImpl.java

```diff
@@ -17,161 +17,177 @@
+        findOptions.oplogReplay(oplogReplay);
         return this;
     }
 
     @Override
-    public FindPublisher<TResult> collation(final Collation collation) {
-        wrapped.collation(collation);
+    public FindPublisher<T> partial(final boolean partial) {
+        findOptions.partial(partial);
         return this;
     }
 
     @Override
-    public FindPublisher<TResult> comment(final String comment) {
-        wrapped.comment(comment);
+    public FindPublisher<T> cursorType(final CursorType cursorType) {
+        findOptions.cursorType(cursorType);
         return this;
     }
 
     @Override
-    public FindPublisher<TResult> hint(final Bson hint) {
-        wrapped.hint(hint);
+    public FindPublisher<T> comment(@Nullable final String comment) {
+        findOptions.comment(comment);
         return this;
     }
 
     @Override
-    public FindPublisher<TResult> hintString(final String hint) {
-        wrapped.hintString(hint);
+    public FindPublisher<T> hint(@Nullable final Bson hint) {
+        findOptions.hint(hint);
         return this;
     }
 
     @Override
-    public FindPublisher<TResult> max(final Bson max) {
-        wrapped.max(max);
+    public FindPublisher<T> hintString(@Nullable final String hint) {
+        findOptions.hintString(hint);
         return this;
     }
 
     @Override
-    public FindPublisher<TResult> min(final Bson min) {
-        wrapped.min(min);
+    public FindPublisher<T> max(@Nullable final Bson max) {
+        findOptions.max(max);
         return this;
     }
 
     @Override
-    public FindPublisher<TResult> returnKey(final boolean returnKey) {
-        wrapped.returnKey(returnKey);
+    public FindPublisher<T> min(@Nullable final Bson min) {
+        findOptions.min(min);
         return this;
     }
 
     @Override
-    public FindPublisher<TResult> showRecordId(final boolean showRecordId) {
-        wrapped.showRecordId(showRecordId);
+    public FindPublisher<T> returnKey(final boolean returnKey) {
+        findOptions.returnKey(returnKey);
         return this;
     }
 
     @Override
-    public FindPublisher<TResult> batchSize(final int batchSize) {
-        wrapped.batchSize(batchSize);
+    public FindPublisher<T> showRecordId(final boolean showRecordId) {
+        findOptions.showRecordId(showRecordId);
         return this;
     }
 
     @Override
-    public FindPublisher<TResult> allowDiskUse(@Nullable final Boolean allowDiskUse) {
-        wrapped.allowDiskUse(allowDiskUse);
+    public FindPublisher<T> allowDiskUse(@Nullable final Boolean allowDiskUse) {
+        findOptions.allowDiskUse(allowDiskUse);
         return this;
     }
 
     @Override
-    public void subscribe(final Subscriber<? super TResult> s) {
-        Publishers.publish(wrapped).subscribe(s);
+    AsyncReadOperation<AsyncBatchCursor<T>> asAsyncReadOperation() {
+        return operations.find(filter, resultClass, findOptions);
```

Comment:
I think this needs to arrange to be a call to `AsyncOperations#findFirst` in the case where first() is called.  Otherwise it won't properly set the limit to 1.

Reply:
Done